core/src/main/java/feign/Client.java

```diff
@@ -161,10 +165,12 @@ HttpURLConnection convertAndSend(Request request, Options options) throws IOExce
       }
 
       if (request.body() != null) {
-        if (contentLength != null) {
-          connection.setFixedLengthStreamingMode(contentLength);
-        } else {
-          connection.setChunkedStreamingMode(8196);
+        if (options.isAllowStreamingMode()) {
```

Comment:
From the documentation, the 2 `set*StreamingMode()` states:

> This method is used to enable streaming of a HTTP request body without internal buffering, when the content length

what if we call this new option `disableInternalBuffering`?

`allowStreamingMode` makes me scratch the head.

Also, this seem to be something very specific to `java.net.HttpURLConnection`.  May be move to `Client.Default`?

or is this something we can enable for ok_http, hc5, apache http 4, google http and so on.

Reply:
Yes, indeed, it's a specific feature for `java.net.HttpURLConnection`, moving the option to `Client.Default` is better. Why didn't I think of that? :-D

I call it `allowStreamingMode` because the word is just literal. I think if one knows the 2 `set*StreamingMode()` methods in `HttpURLConnection`, so he can understand what streaming mode is.

After moving `allowStreamingMode` into `Client.Default`, it will be related to that specific implementation. Then it may be easier to understand.

Reply:
yeah, in my mind, `allowStreamingMode` doesn't say much, like `allowFlushingMode`... I have no idea what either mean.

now, `disableInternalBuffering`, at least for me, has an exact meaning... client has some kind of internal buffer, and it's being turned off.

Anyway, I won't be picky about this, do what you feel best about.

Reply:
Discussed with my colleagues. Now we all agree `allowStreamingMode` is not so good, and we checked the source code found the buffering only happens in writing request output stream, so the final desicion is call it `disableRequestBuffering`

Reply:
Sound great